File: BinarySearch.py
import math

def binary_search (arr,element):
    low = 0
    high = len(arr) - 1

    while low <= high:
        pointer = math.floor((low + high) / 2)
        value = arr[pointer]

        if value == element:
            return True
        elif element < value:
            high = pointer - 1
        else: 
            low = pointer + 1

    return False



arr = [1,2,3,4,5,6,7,8,9,10]

print(binary_search(arr,10))
print(binary_search(arr,1))
print(binary_search(arr,7))
print(binary_search(arr,2))

File: src/BinarySearch.ts
export const BinarySearch = (array: number[], valueToFind: number): boolean => {
  let leftPointer = 0;
  let rightPointer = array.length - 1;

  do {
    const pointer = Math.floor((leftPointer + rightPointer) / 2);
    const currentValue = array[pointer];

    if (currentValue == valueToFind) {
      return true;
    } else if (currentValue > valueToFind) {
      rightPointer = pointer - 1;
    } else {
      leftPointer = pointer + 1;
    }
  } while (leftPointer <= rightPointer);

  return false;
};

BinarySearch([1, 2, 3, 4, 5], 5);
BinarySearch([1, 2, 3, 4, 5], 2);
BinarySearch([1, 2, 3, 4, 5], 3);
BinarySearch([1, 2, 3, 4, 5], 1);
